### PR TITLE
Improve Python conflicts

### PR DESCRIPTION
**Pull request by AntoinePrv**
A heuristic to better handle the (almost) cyclic conflict due to the reversal of the dependencies bewteen `python` and `python_abi`.
The heuristic looks for this pattern and tries to simplify it.

The heuristics required removing nodes/conflicts, which led to most of the changes in this PR.

- [x] Based on #2314

### CODE REVIEW

File: libmamba/include/mamba/core/satisfiability_error.hpp [hunk 1]
```diff
@@ -70,31 +77,14 @@ namespace mamba
         };
         struct PackageNode : PackageInfo
```
Comment (JohanMabille) [inline libmamba/include/mamba/core/satisfiability_error.hpp:78]: Why keeping empty types? If you find the name more expressive, you can alias the type (although it won't prevent to pass the real type to functions accepting the alias).

File: libmamba/include/mamba/core/satisfiability_error.hpp [hunk 2]
```diff
@@ -70,31 +77,14 @@ namespace mamba
...
         {
-            std::optional<SolverRuleinfo> problem_type = {};
-
-            PackageNode(const PackageNode&) = default;
-            PackageNode(PackageNode&&) noexcept = default;
-            PackageNode& operator=(const PackageNode&) = default;
-            PackageNode& operator=(PackageNode&&) noexcept = default;
         };
         struct UnresolvedDependencyNode : MatchSpec
```
Comment (JohanMabille) [inline libmamba/include/mamba/core/satisfiability_error.hpp:81]: Same here.

File: libmamba/include/mamba/core/satisfiability_error.hpp [hunk 3]
```diff
@@ -70,31 +77,14 @@ namespace mamba
...
         {
-            SolverRuleinfo problem_type;
-
-            UnresolvedDependencyNode(const UnresolvedDependencyNode&) = default;
-            UnresolvedDependencyNode(UnresolvedDependencyNode&&) noexcept = default;
-            UnresolvedDependencyNode& operator=(const UnresolvedDependencyNode&) = default;
-            UnresolvedDependencyNode& operator=(UnresolvedDependencyNode&&) noexcept = default;
         };
         struct ConstraintNode : MatchSpec
```
Comment (JohanMabille) [inline libmamba/include/mamba/core/satisfiability_error.hpp:84]: Same here.

File: libmamba/include/mamba/core/satisfiability_error.hpp (hunk 1 above)
Comment (AntoinePrv) [reply libmamba/include/mamba/core/satisfiability_error.hpp:78]: Strong typing. They are put in a `std::variant` so they cannot be the same type.

Comment (JohanMabille) [reply libmamba/include/mamba/core/satisfiability_error.hpp:78]: Oh, indeed.

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**